site/content/en/docs/user/stages-configuration.md

```diff
@@ -138,6 +139,43 @@ if a "Change Stage" is matched. The update action itself consequently causes a n
 However, this event-driven approach to applying Stages has a limitation: `kwok` won’t apply a Stage until a new event associated with that resource is received. To address the limitation,
 users can utilize the `immediateNextStage` field to make the controller apply Stages immediately rather than waiting for an event pushed from the apiserver.
 
+## How Delay is Calculated
+
+The delay time of applying a Stage is obtained by adding a constant time period and a randomized interval,
+which will be denoted by *duration* and *jitter* respectively in the following.
+
+- `durationMilliseconds`: provides *duration*.
+- `jitterDurationMilliseconds`: calculates *jitter* by *random*(`jitterDurationMilliseconds`-*duration*).
+  It shall be larger than `durationMilliseconds` if you want to inject jitter to the delay.
+  Otherwise, it will be used directly as the delay time without any randomization.
+
+`kwok` also provides other fields to flexibly handle more situations. They are used to extract
+and parse a RFC3339 timestamp field of a resource to help determine the delay time dynamically.
+
+- `durationFrom`: calculates *duration* by `durationFrom`-*now*
+- `jitterDurationFrom`: calculates *jitter* by *random*(`jitterDurationFrom`-*now*-*duration*),
+  where *now* is the timestamp when the delay starts.
+
+{{< hint "info" >}}
+`durationFrom` and `jitterDurationFrom` have a higher priority than `durationMilliseconds`
+and `jitterDurationMilliseconds` if both are set.
+{{< /hint >}}
+
+Let’s explain a little bit about the motivation behind these two advanced fields.
+But before that, for a better understanding, we briefly describe how kubelet "delete" a pod from a node here.
+Basically when you run `kubectl delete pod`, apiserver will not directly delete the corresponding pod resource
+from etcd when receiving the deletion request, but instead set the `.metadata.deletionTimestamp` field to the time
+the request was issued plus a short period: `metadata.deletionGracePeriodSeconds` (default 30s).
+kubelet starts to send a `TERM` signal to the container main process once a non-null `metadata.deletionTimestamp`
+of a pod is detected. The main process is then terminated using the `KILL` signal if the `.metadata.deletionTimestamp`
+expires before the process stops by itself. After all the containers in the pod have stopped running, kubelet will send
+a force deletion request to apiserver to delete the pod object from etcd. To simulate such situation, you can let 
+`jitterDurationFrom` of a "Delete Stage" (`next.delete: true`) point to `metadata.deletionTimestamp`. This will make
+the deletion operation happen at a random moment before `metadata.deletionTimple` expires.  
```

Comment:
It feels clearer to describe the process step by step

Here are the steps to remove a pod in Kubernetes:

1. Execute the command `kubectl delete pod`. The apiserver receives the deletion request but does not immediately remove the corresponding pod resource from etcd.
2. The apiserver sets the `metadata.deletionTimestamp` field to the time the request was issued plus a short period, defined by `metadata.deletionGracePeriodSeconds` (default 30s).
3. The kubelet detects a non-null `metadata.deletionTimestamp` for a pod and starts to send a `TERM` signal to the main process of the container.
4. If the `metadata.deletionTimestamp` expires before the process stops by itself, the main process is then terminated using the `KILL` signal.
5. After all the containers in the pod have stopped running, the kubelet sends a force deletion request to the apiserver.
6. The apiserver removes the pod object from etcd.

To simulate this situation, you can set `jitterDurationFrom` of a "Delete Stage" (`next.delete: true`) to point to `metadata.deletionTimestamp`. This will cause the deletion operation to occur at a random moment before `metadata.deletionTimestamp` expires.

Reply:
Sure. Thanks!